pkg/proxy/service.go

```diff
@@ -146,10 +146,14 @@ func (sct *ServiceChangeTracker) newBaseServiceInfo(port *v1.ServicePort, servic
 		topologyKeys:           service.Spec.TopologyKeys,
 	}
 
+	loadBalancerSourceRanges := make([]string, len(service.Spec.LoadBalancerSourceRanges))
+	for i, sourceRange := range service.Spec.LoadBalancerSourceRanges {
+		loadBalancerSourceRanges[i] = strings.TrimSpace(sourceRange)
+	}
```

Comment:
Does this solve the actual crashloop though? If the input is invalid

Reply:
Thanks for double checking. I did some more thorough testing to ensure this would fix the problem and have given a more detailed explanation below.